### PR TITLE
Feature product blacklist storage

### PR DESCRIPTION
**Pull request by romainruaud**
Here is the backend part : adding column to both search position and virtual category product position tables.

### CODE REVIEW

**Comment by romainruaud** [comment]
PR updated : 

- position field is now nullable
- ensure creating the tables with proper structure on raw install, process table modification only on upgrade.

File: src/module-elasticsuite-catalog/Setup/CatalogSetup.php [hunk 1]
```diff
@@ -445,9 +445,16 @@ public function createSearchPositionTable(SchemaSetupInterface $setup)
             'position',
             \Magento\Framework\DB\Ddl\Table::TYPE_INTEGER,
             null,
-            ['nullable' => false, 'default' => '0'],
+            ['nullable' => true, 'default' => '0'],
```
Comment (afoucret) [inline src/module-elasticsuite-catalog/Setup/CatalogSetup.php]: Nullable field should not have a default value.

File: src/module-elasticsuite-virtual-category/Setup/VirtualCategorySetup.php [hunk 2]
```diff
@@ -183,9 +183,16 @@ public function createPositionTable(SchemaSetupInterface $setup)
                 'position',
                 \Magento\Framework\DB\Ddl\Table::TYPE_INTEGER,
                 null,
-                ['nullable' => false, 'default' => '0'],
+                ['nullable' => true, 'default' => '0'],
```
Comment (afoucret) [inline src/module-elasticsuite-virtual-category/Setup/VirtualCategorySetup.php]: Nullable field should not have a default value

File: src/module-elasticsuite-virtual-category/Setup/VirtualCategorySetup.php [hunk 3]
```diff
@@ -205,4 +212,42 @@ public function createPositionTable(SchemaSetupInterface $setup)
...
+     */
+    public function setNullablePositionColumn(SchemaSetupInterface $setup)
+    {
+        $setup->getConnection()->modifyColumn(
+            $setup->getTable('smile_virtualcategory_catalog_category_product_position'),
+            'position',
+            [
+                'type'     => \Magento\Framework\DB\Ddl\Table::TYPE_INTEGER,
+                'nullable' => true,
+                'default'  => 0,
```
Comment (afoucret) [inline src/module-elasticsuite-virtual-category/Setup/VirtualCategorySetup.php]: Nullable field should not have a default value.

File: src/module-elasticsuite-catalog/Setup/CatalogSetup.php [hunk 4]
```diff
@@ -468,6 +475,44 @@ public function createSearchPositionTable(SchemaSetupInterface $setup)
...
+     */
+    public function setNullablePositionColumn(SchemaSetupInterface $setup)
+    {
+        $setup->getConnection()->modifyColumn(
+            $setup->getTable('smile_elasticsuitecatalog_search_query_product_position'),
+            'position',
+            [
+                'type'     => \Magento\Framework\DB\Ddl\Table::TYPE_INTEGER,
+                'nullable' => true,
+                'default'  => 0,
```
Comment (afoucret) [inline src/module-elasticsuite-catalog/Setup/CatalogSetup.php]: Nullable field should not have a default value.

File: src/module-elasticsuite-virtual-category/Setup/VirtualCategorySetup.php [hunk 5]
```diff
@@ -183,9 +183,16 @@ public function createPositionTable(SchemaSetupInterface $setup)
...
                 null,
-                ['nullable' => false, 'default' => '0'],
+                ['nullable' => true, 'default' => '0'],
                 'Position'
             )
+            ->addColumn(
+                'is_blacklisted',
+                \Magento\Framework\DB\Ddl\Table::TYPE_BOOLEAN,
+                null,
+                ['nullable' => true, 'default' => '0'],
```
Comment (afoucret) [inline src/module-elasticsuite-virtual-category/Setup/VirtualCategorySetup.php]: This field should not be nullable.

**Comment by romainruaud** [comment]
Ok, I updated the PR.

Thank you

**Pull request merged**